src/profile-logic/import/instruments/instruments.js

```diff
@@ -0,0 +1,746 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+//types
+import type { Profile } from '../../../types/profile';
+
+//utils
+import {
+  getEmptyProfile,
+  getEmptyThread,
+} from '../../../profile-logic/data-structures';
+import BinaryPlistParser, { UID } from './BinaryPlistParser';
+import BinReader from './BinReader';
+
+// TODO make helpers.js and move the appropriate helper functions into it
+// TODO add the missing return types in the functions
+// TODO I have used console.log() statements for debug purpose, I will remove them later
+// TODO add Root frame at the top
+// TODO fill remaining meta data inside profile
+
+let fileReader;
+
+function parseBinaryPlist(bytes) {
+  const text = 'bplist00';
+  for (let i = 0; i < 8; i++) {
+    if (bytes[i] !== text.charCodeAt(i)) {
+      throw new Error('File is not a binary plist');
+    }
+  }
+
+  return new BinaryPlistParser(
+    new DataView(bytes.buffer, bytes.byteOffset, bytes.byteLength)
+  ).parseRoot();
+}
+
+function decodeUTF8(bytes: Uint8Array): string {
+  let text = String.fromCharCode.apply(String, bytes); // eslint-disable-line prefer-spread
```

Comment:
Is there a reason not to use spread operator here?

Reply:
<img width="601" alt="Screenshot 2019-07-12 at 1 06 58 AM" src="https://user-images.githubusercontent.com/16744065/61079964-9fa91a80-a441-11e9-9ffa-4e4e7537af59.png">

I was getting this error if I don't use it.

Reply:
Can you please share the actual line you were using? Thanks!

Reply:
```  let text = String.fromCharCode(...bytes);```
I was using this line before. But as @mstange suggested, implementation of this function is changed.